packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1424,6 +1424,36 @@ class Transform extends SingleChildRenderObjectWidget {
         assert(scale == null || (scaleX == null && scaleY == null), "If 'scale' is non-null then 'scaleX' and 'scaleY' must be left null"),
         transform = Matrix4.diagonal3Values(scale ?? scaleX ?? 1.0, scale ?? scaleY ?? 1.0, 1.0);
 
+  /// Creates a widget that flips/mirrors its child along the 2D plane.
+  ///
+  /// If `flipX` is true, the child widget will be flipped horizontally. Defaults to false.
+  ///
+  /// If `flipY` is true, the child widget will be flipped vertically. Defaults to false.
+  ///
+  /// If both are true, the child widget will be flipped both vertically and horizontally.
+  ///
+  /// {@tool snippet}
+  ///
+  /// This example flips the text horizontally.
+  ///
+  /// ```dart
+  /// Transform.flip(
+  ///   flipX: true,
+  ///   child: const Text('Horizontal Flip'),
+  /// )
+  /// ```
+  /// {@end-tool}
+  Transform.flip({
+      super.key,
+      bool flipX = false,
+      bool flipY = false,
+      this.origin,
+      this.alignment = Alignment.center,
+      this.transformHitTests = true,
+      this.filterQuality,
+      super.child,
+    }) : transform = Matrix4.diagonal3Values(flipX ? -1.0 : 1.0, flipY ? -1.0 : 1.0, flipX ^ flipY ? -1.0 : 1.0);
```

Comment:
What effect does the -1 z value have?

Reply:
I applied matrix rotation and got this. So I thought I should preserve the value. It does not however affect the appearance as far as I noticed
<img width="493" alt="CleanShot 2022-12-10 at 03 30 34@2x" src="https://user-images.githubusercontent.com/71540402/206841221-bd518272-342a-47bf-a0bc-b00c442cca0a.png">

Reply:
You shouldn't need to supply -1 in Z, you're just doing a 2D flip. I'd just always set it to 1.

Reply:
Making Z constant to reduce unnecessary computation 👍